src/resources/coupons-resource.js

```diff
@@ -46,7 +46,7 @@ export default function CouponsResource({apiHandler}) {
             return apiHandler.post(`coupons-redemptions`, data);
         },
 
-        setExpiry({redemptionCode, data}) {
+        setExpiration({redemptionCode, data}) {
```

Comment:
This means we need to make the changes on Recomm and tell merchants that are using this.

Reply:
Yeah, although it's insulated behind a major version change. There are several breaking changes in this PR; my thinking was we can do a major release which brings the JS-SDK to be fully consistent with `api-definitions`

But I'm very interested in getting feedback on that! This PR was easy to throw together while working on the SDK docs, so I consider it somewhat of a proposal

Reply:
Gotcha, thanks! Same here I would love to hear from others about this subject

Reply:
So looking through the API logs, it doesn't seem like this endpoint has ever been meaningfully been used on live or sandbox, it's almost entirely 4xx errors. It's not present in the Recomm codebase either. 

Maybe we can just drop it entirely?

Reply:
I'd like to see us start to use a two-step approach to removing and refactoring endpoint names like this.

one major release to throw a depreciation notice, then after some time another major release to remove it.

the old function could look something like this

```
setExpiry() {
  console.warn('"setExpiry" function is deprecated and will be remove with the next major release version. Please use "setExpiration" instead. See docs www.docs.rebilly.com');
  setExpiration(arguments);
}
```

Reply:
For the majority of situations I completely agree with an intermediate deprecation step, but I don't believe anyone has ever actually used the `setExpiry` method anywhere. The endpoint itself isn't really used at all, let alone SDK method.

That's why I think a rename or outright removal in a single major version would be okay here.

Reply:
Basically I agree with @mathewsonke , renaming it is easy if no one is using it

Reply:
> This means we need to make the changes on Recomm and tell merchants that are using this.

@cesarlevel no. This means Recomm has to make changes only if it upgrades to the next major version of the js-sdk. There is a big nuance there, and it's really important to see it.

I actually want to remove the js-sdk entirely. More on that in an RFC.